src/main/java/Domain/Game/Team.java

```diff
@@ -284,6 +335,165 @@ public void setTeamName(String testName) {
         this.teamName = testName;
     }
 
+    /**
+     *Getter Stadiums
+     * @return list of Stadiums
+     */
+    public List<Stadium> getStadiums() {
+        return stadiums;
+    }
+
+    /**
+     *
+     * @param stadium
+     * add stadium to team
+     * @return true if stadium added successfully!
+     */
+    public boolean addStadium(Stadium stadium) {
+        if (!this.stadiums.contains(stadium)) {
+            this.stadiums.add(stadium);
+        }
+        return true;
+    }
+
+    /**
+     *
+     * @param stadium
+     * remove stadium to team
+     * @return true if stadium removed successfully!
+     */
+    public boolean removeStadium(Stadium stadium) {
+        if(this.stadiums.contains(stadium))
+        {
+            this.stadiums.remove(stadium);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Each team has a lot of Assets {@link TeamAsset},
+     * @return a list of all the team assets
+     */
+    public List<Asset> getAllAssets() {
+
+        List<Asset> allTeamAssets = new ArrayList<>();
+        allTeamAssets.addAll(this.teamPlayers);
+        allTeamAssets.addAll(this.teamCoaches);
+        allTeamAssets.addAll(this.teamManagers);
+        allTeamAssets.addAll(this.stadiums);
+        return allTeamAssets;
+    }
+
+    /**
+     * Checks whether or not the teamOwner is owner if this team
+     * @param teamOwner the team owner to validate with
+     * @return true if teamOwner is this team wner.
+     */
+    public boolean isTeamOwner(TeamOwner teamOwner) {
+        if(teamOwner != null && teamOwners.contains(teamOwner))
+        {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * General Function to add an asset to the team,
+     * receives an assetName a teamOwner and an AssetType
+     * @param assetName the assetName suppose to be the Username of the {@link SystemUser} or the name of the {@link Stadium}
+     * @param teamOwner the {@link TeamOwner} of this team
+     * @param assetType the {@link TeamAsset} of the user to add to the team
+     * @return true if the asset was added successfully
+     * @throws UserNotFoundException
+     * @throws StadiumNotFoundException
+     */
+    public boolean addAsset(String assetName, TeamOwner teamOwner, TeamAsset assetType)
+            throws UserNotFoundException,StadiumNotFoundException {
+
+        Asset teamAsset;
+        SystemUser assetUser;
+        if(!assetType.equals(TeamAsset.STADIUM)) {
+            assetUser = EntityManager.getInstance().getUser(assetName);
+            if (assetUser == null) {
+                throw new UserNotFoundException("Could not find user " + assetName);
```

Comment:
The "if" is talking about stadium, but inside you are dealing with a user (and throwing a "could not find user" exception).  Maybe you did it accidentally?